Fixes #141: Handle node (#3)
utnaf/neo4j-spark-connector: doc/docs/modules/ROOT/pages/v2.adoc

> = Quick Start

Reviewer: elimina questo file e metti tutto dentro quickstart.adoc

[abstract]
--
Get started fast for common scenarios, using Neo4j Streams plugin or Kafka Connect plugin
--

== Neo4j Spark Connector Plugin

=== Overview

> TK Overview

Reviewer: Scrivici due righe due senno ce le fa mettere lui

=== Getting Started

> TK Provide short description of the example code

Reviewer: Anche qui

```scala
val df = spark.read.format("neo4j")
        .option("url", "bolt://localhost:7687")

>         .option("node.labels", "Person")

Reviewer: dovrebbe essere così:
```suggestion
        .option("labels", "Person")
```

        .load()
```

=== Neo4j Options

Spark method `option` is used to configure the Neo4j Connector.

```scala
val df = spark.read.format("neo4j")
        .option("url", "bolt://localhost:7687")
        .option("authentication.type", "basic")
        .option("authentication.basic.username", "myuser")
        .option("authentication.basic.password", "neo4jpassword")

>         .option("node.labels", "Person")

Reviewer: come sopra

|``
|Yes^*^

|`schema.flatten.limit`
|Number of records to be used to create the Schema (only if APOC are not installed)
|`10`
|No

|===

^*^ Just one of the options can be specified.

== Read Data

Reading data from a Neo4j Database can be done in 3 ways:

 * with a Cypher query
 * with a set of node Labels 
 * by specifying a relationship

=== Considerations on the schema

Spark works with data in a tabular fixed schema. To accomplish this Neo4j Connector has a schema infer system that creates the schema based on the data requested for the read. Each read data method has is own strategy to create it, that will be explained it each section.

TK list of supported data types

=== Read data by Node Labels

You can both specify a single label, like this example
```scala
val df = spark.read.format("neo4j")
        .option("url", "bolt://localhost:7687")
        .option("node.labels", "Person")
        .load()

df.show()
```

Multiple labels can be specified, separated by `:`
```scala
val df = spark.read.format("neo4j")
        .option("url", "bolt://localhost:7687")
        .option("node.labels", "Person:Customer:Admin")
        .load()

df.show()
```

> Remember that the first Label is considered to be the primary label.

Reviewer: ```suggestion
[NOTE]
Remember that the first Label is considered to be the primary label.
```

> When reading data with this method, the result set will contain all the columns contained in the nodes, plus 2 additional columns.

Reviewer: ```suggestion
When reading data with this method, the Dataframe will contain all the fields contained in the nodes, plus 2 additional columns.
```

cerchiamo di usare i termini di Spark/Neo4j
